### DISCUSSION TITLE
Error overlay is shown, even when there's an error boundary

### DISCUSSION THREAD

**Discussion by Janpot** [vercel/next.js#44926] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Reproduction in the following sandbox: https://codesandbox.io/s/unruffled-darkness-2xqhi1?file=/pages/index.js

### Describe the Bug

When an error is thrown in a component that is rendered under an error boundary, the Next.js error overlay is displayed, even though the error is handled. Showing the overlay is very disruptive and a false alarm.

### Expected Behavior

No overlay is shown

### To Reproduce

Open the [code sandbox](https://codesandbox.io/s/unruffled-darkness-2xqhi1?file=/pages/index.js) and click the button.

**Comment by timneutkens** [comment]
This is intentional as per https://github.com/facebook/create-react-app/issues/6530#issuecomment-662914362

**Comment by Janpot** [comment]
That's a curious line of reasoning. It's like saying "hey, we're going to ignore your `try`/`catch` statements just in case you didn't intentionally put them there".

Doesn't suspense for data fetching rely heavily on error boundaries? How will this ever play nice with error overlay?

**Comment by jankaifer** [comment]
~~Since this is not really `next.js` specific, I'll close this as a duplicate of https://github.com/facebook/create-react-app/issues/6530~~

EDIT: my bad it's not the `react` repo but CRA.

**Comment by jankaifer** [comment]
Either way, this is not a bug and we think it is the right thing to do.
As Dan said in https://github.com/facebook/create-react-app/issues/6530#issuecomment-662914362 

I'll move this into discussions where we can discuss if there are some valid use-cases that would benefit from this.

**Comment by wfl-junior** [comment]
How can I at least opt out of showing error overlays for errors that are caught by an ErrorBoundary?

**Comment by imclint21** [comment]
Any news on this? This modal is really disturbing when we want to catch errors with error boundaries.

<img width="593" alt="image" src="https://github.com/vercel/next.js/assets/5221349/80fe98b6-cd54-48e6-ab2e-6a84be408e34">
